modules/data-entry/src/main/frontend/src/questionnaireEditor/Fields.jsx

```diff
@@ -91,10 +94,15 @@ let Fields = (props) => {
      return keys;
   };
 
-  return edit ?
-    Object.entries(JSON).map(([key, value]) => displayEditField(key, value))
-    :
-    Object.entries(JSON).map(([key, value]) => (data[key] ? displayStaticField(key, value) : ''));
+  // Note that we remove the //REQUIRED field, which just indicates which fields are mandatory
+  return <FieldsProvider>
+      {
+        edit ?
+          Object.entries(JSON).filter(([key, _]) => key != "//REQUIRED").map(([key, value]) => displayEditField(key, value))
+        :
+          Object.keys(getAllKeys(JSON)).filter(([key, _]) => key != "//REQUIRED").map(key => (data[key] ? displayStaticField(key) : ''))
```

Comment:
Should this be made a bit more generic and exclude anything starting with `//`?

```suggestion
          Object.entries(JSON).filter(([key, _]) => !key.startsWith("//")).map(([key, value]) => displayEditField(key, value))
        :
          Object.keys(getAllKeys(JSON)).filter(([key, _]) => !key.startsWith("//")).map(key => (data[key] ? displayStaticField(key) : ''))
```

Also, the last line is buggy, it should probably be:

```suggestion
          Object.entries(JSON).filter(([key, _]) => !key.startsWith("//")).map(([key, value]) => displayEditField(key, value))
        :
          Object.keys(getAllKeys(JSON)).filter((key) => !key.startsWith("//")).map(key => (data[key] ? displayStaticField(key) : ''))
```

But even like this, I'm not sure it will work properly since `displayStaticField` kind of requires a value, which is no longer passed. The only consequence so far seems to be that markdown fields are not going to be rendered: in the questionnaire editor, edit any question, add a formatted description, save -> the description is not formatted in the questionnaire editor.